modules/widgets/src/fullscreen-widget.tsx

```diff
@@ -0,0 +1,141 @@
+/* global document */
+import {_deepEqual as deepEqual} from '@deck.gl/core';
+import type {Deck, Widget, WidgetPlacement} from '@deck.gl/core';
+import {h, render} from 'preact';
+import {IconButton} from './components';
+
+interface FullscreenWidgetProps {
+  id: string;
+  viewId?: string | null;
+  placement?: WidgetPlacement;
+  /**
+   * A [compatible DOM element](https://developer.mozilla.org/en-US/docs/Web/API/Element/requestFullScreen#Compatible_elements) which should be made full screen.
+   * By default, the map container element will be made full screen.
+   */
+  /* eslint-enable max-len */
+  container?: HTMLElement;
+  label?: string;
+  style?: Partial<CSSStyleDeclaration>;
+  className?: string;
+}
+
+export class FullscreenWidget implements Widget<FullscreenWidgetProps> {
+  id = 'fullscreen';
+  props: FullscreenWidgetProps;
+  placement: WidgetPlacement = 'top-left';
+  viewId = null;
+
+  deck?: Deck;
+  element?: HTMLDivElement;
+
+  fullscreen: boolean = false;
+
+  constructor(props: FullscreenWidgetProps) {
+    this.id = props.id || 'fullscreen';
+    this.viewId = props.viewId || null;
```

Comment:
What does it mean to assign the full screen widget to a specific view? My understanding is that it will always full screen the whole canvas.

Reply:
I also think this is a little odd, but I understood it as `WidgetManager` offsets position of the widget to be inside the view of choice. Fullscreen behavior will always apply to a whole container, be it `canvas.parentElement` or user provided `container` prop.

Reply:
I think that will only confuse the user more. I suggest you remove this prop for now and add it back if it is needed.

Reply:
Agreed.. just cuz we can position it anywhere doesn't mean we should